cpp/src/arrow/record_batch.h

```diff
@@ -254,6 +255,16 @@ class ARROW_EXPORT RecordBatch {
   /// \return Status
   virtual Status ValidateFull() const;
 
+  /// \brief Return a top-level sync event object for this record batch
```

Comment:
Can we mark this EXPERIMENTAL until we're sure this is the right API at the right place?